### PR TITLE
[WIP] Clean booking rspec

### LINKED ISSUE #20: Fix booking quantity validation
Fix the logic of the enough_items validation of booking model

### PR DESCRIPTION
**Pull request by NicholasCF**
Fixes #20. 

Changed the quantity validation.
Checked with rspec, but the test name and implementation may be lacking.

### CODE REVIEW

**Comment by NicholasCF** [comment]
There's an issue with the counting, working on it

### Edit
Turns out includes is used only for associations, while for eager loading attributes, select must be used.
I also forgot that ActiveRecord is lazily evaluated, so error lines are not what they seemed

**Comment by NicholasCF** [comment]
The quantity validation now correctly bases itself on the total quantity of reserved item, instead of the total number of bookings

**Comment by NicholasCF** [comment]
I added the ability for curr to jump to the next start_time, instead of incrementing in 30 minutes

File: spec/controllers/bookings_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    it 'normal' do
+      get :index
+      should respond_with :ok
+    end
+  end
+
+  describe 'POST bookings#create' do
+    before do
+      user = create(:user)
+      user.add_role(:technical)
```
Comment (sreycodes) [inline spec/controllers/bookings_controller_spec.rb]: Technically add_role shouldn't work. Don't do this here

File: spec/controllers/bookings_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    end
+
+    it 'should change the description and redirect' do
+      expect do
+        patch :update, params: { id: @booking.id, booking: {description: "I'm super duper Rich"} }
+      end.to change { Booking.find(@booking.id).description }
+        .to("I'm super duper Rich")
+      should redirect_to bookings_path
+    end
+
```
Comment (sreycodes) [inline spec/controllers/bookings_controller_spec.rb:195]: Add a test changing to a quantity greater than the item.quantity and make sure it doesn't work

**Review by sreycodes — CHANGES_REQUESTED** [review]

**Review by sreycodes — APPROVED** [review]

**Pull request merged**